docs/aurora/data-science/inference/vllm.md

```diff
@@ -0,0 +1,102 @@
+
+For small models that fit within a single tile's memory (64 GB), no additional configuration is required to serve the model. Simply set `TP=1` (Tensor Parallelism). This configuration ensures the model is run on a single tile without the need for distributed setup.
+
+To utilize multiple tiles for larger models (`TP>1`), a more advanced setup is necessary. This involves configuring a Ray cluster and setting the `ZE_FLAT_DEVICE_HIERARCHY` environment variable.
+```bash
+export ZE_FLAT_DEVICE_HIERARCHY=FLAT
+
+export VLLM_HOST_IP=$(getent hosts $(hostname).hsn.cm.aurora.alcf.anl.gov | awk '{ print $1 }' | tr ' ' '\n' | sort | head -n 1)
+export tiles=12
+ray --logging-level debug start --head --verbose --node-ip-address=$VLLM_HOST_IP --port=6379 --num-cpus=64 --num-gpus=$tiles&
+```
+!!! info "DLPack notes on Aurora"
+	- The number of GPU devices visible on each node depends on the `ZE_FLAT_DEVICE_HIERARCHY` environment variable. When ZE_FLAT_DEVICE_HIERARCHY=FLAT, the system exposes 12 devices (tiles).
+    - The --num-gpus parameter in the Ray cluster setup command should match the number of tiles you intend to use on each node.
+
+
+## Serve Small Models 
+
+#### Using Single Tile
+
+Following serves `meta-llama/Llama-2-7b-chat-hf` on single tile of single node. 
+```bash
+vllm serve meta-llama/Llama-2-7b-chat-hf --port 8000 --device xpu --dtype float16
+```
+
+#### Using Multiple Tiles
+
+Following sets up ray cluster and servers `meta-llama/Llama-2-7b-chat-hf` on 8 tiles on single node. 
+```bash
+export VLLM_HOST_IP=$(getent hosts $(hostname).hsn.cm.aurora.alcf.anl.gov | awk '{ print $1 }' | tr ' ' '\n' | sort | head -n 1)
+ray --logging-level debug start --head --verbose --node-ip-address=$VLLM_HOST_IP --port=6379 --num-cpus=64 --num-gpus=8&
+vllm serve meta-llama/Llama-2-7b-chat-hf --port 8000 --tensor-parallel-size 8 --device xpu --dtype float16 --trust-remote-code
+```
+
+
+## Serve Medium Models 
+
+#### Using single Node
+
+Following sets up ray cluster and servers `meta-llama/Llama-3.3-70B-Instruct` on 8 tiles on single node. 
+```bash
+export VLLM_HOST_IP=$(getent hosts $(hostname).hsn.cm.aurora.alcf.anl.gov | awk '{ print $1 }' | tr ' ' '\n' | sort | head -n 1)
+ray --logging-level debug start --head --verbose --node-ip-address=$VLLM_HOST_IP --port=6379 --num-cpus=64 --num-gpus=8&
+vllm serve meta-llama/Llama-3.3-70B-Instruct --port 8000 --tensor-parallel-size 8 --device xpu --dtype float16 --trust-remote-code
+```
+
+#### Using Multiple Nodes
+
+Use [setup_ray_cluster.sh](./setup_ray_cluster.sh) script to setup ray cluster across nodes. 
+
+Following serves `meta-llama/Llama-3.3-70B-Instruct` using 2 nodes with `TP=8` and `PP=2` 
+
+```bash
+vllm serve meta-llama/Llama-3.3-70B-Instruct --port 8000 --tensor-parallel-size 8 --pipeline-parallel-size 2 --device xpu --dtype float16 --trust-remote-code
+```
+
+## Serve Large Models 
+
+Use [setup_ray_cluster.sh](./setup_ray_cluster.sh) script to setup ray cluster across nodes. 
```

Comment:
```suggestion
Use [setup_ray_cluster.sh](./setup_ray_cluster.sh) script to setup a Ray cluster across nodes. 
```

And change the link to a path in `../../../GettingStarted/` after moving the script?

Reply:
@sraskar Can you also provide the run_ray_cluster.sh script. This should run ray head and ray workers across nodes. We also would need to specify how to fetch nodes through qsub. I can work on it if you provide the run script